### PR TITLE
#1921: Hovering over some functional groups crops the expanded view of it

### PR DESCRIPTION
**Pull request by KonstantinEpam23**
Closes: #1921 #1911  #1922

### CODE REVIEW

File: packages/ketcher-react/src/script/editor/Editor.ts [hunk 1]
```diff
@@ -210,9 +210,9 @@ class Editor implements KetcherEditor {
 
     const tool = new toolMap[name](this, opts)
 
-    // hide icon if not AtomToll chosen
-    if (name !== 'atom') {
-      this.render.paper.getById('atomHoverIcon')?.hide()
+    const isAtomToolChosen = name === 'atom'
```
Comment (KonstantinEpam23) [inline packages/ketcher-react/src/script/editor/Editor.ts:213]: reverted

File: packages/ketcher-react/src/script/editor/Editor.ts [hunk 2]
```diff
@@ -363,25 +363,25 @@ class Editor implements KetcherEditor {
 
     if (!event) return
 
-    const myCi = this.findItem(event, ['sgroups', 'functionalGroups'])
+    const checkFunctionGroupTypes = ['sgroups', 'functionalGroups']
+    const myCi = this.findItem(event, checkFunctionGroupTypes)
     if (myCi) {
-      if (myCi.map === 'sgroups' || myCi.map === 'functionalGroups') {
+      if (checkFunctionGroupTypes.includes(myCi.map)) {
```
[the remark this answers is not in the document]
Comment (KonstantinEpam23) [reply packages/ketcher-react/src/script/editor/Editor.ts:369]: You are absolutely right, I think I have it a a leftover from previous iteration of working on this. Removed.

**Comment by KonstantinEpam23** [comment]
Closes: https://github.com/epam/ketcher/issues/1920 https://github.com/epam/ketcher/issues/1921 https://github.com/epam/ketcher/issues/1922

File: packages/ketcher-core/src/application/render/raphaelRender.js [hunk 3]
```diff
@@ -226,21 +226,27 @@ Render.prototype.update = function (force = false, viewSz = null) {
...
       const sz1 = bb.sz()
-      const marg = this.options.autoScaleMargin
-      const mv = new Vec2(marg, marg)
+      const { autoScaleMargin, rescaleAmount, maxBondLength } = this.options
+      const mv = new Vec2(autoScaleMargin, autoScaleMargin)
       const csz = viewSz
-      if (marg && (csz.x < 2 * marg + 1 || csz.y < 2 * marg + 1)) {
+      if (
+        autoScaleMargin &&
+        (csz.x < 2 * autoScaleMargin + 1 || csz.y < 2 * autoScaleMargin + 1)
```
Comment (ansivgit) [inline packages/ketcher-core/src/application/render/raphaelRender.js:234]: Maybe it makes sense to put 2 * autoScaleMargin in a variable with a meaningful name?

Comment (KonstantinEpam23) [reply packages/ketcher-core/src/application/render/raphaelRender.js:234]: to be honest, I still dont completely understand how exactly this works xP

**Review by Nitvex — CHANGES_REQUESTED** [review]
At the moment, these changes crash the application after adding some salt or solvent to canvas:
image

**Review by Nitvex — APPROVED** [review]

**Pull request merged**